homeassistant/components/mqtt/device_trigger.py

```diff
@@ -173,9 +175,15 @@ async def update_trigger(self, config: ConfigType) -> None:
         # Unsubscribe+subscribe if this trigger is in use and topic has changed
         # If topic is same unsubscribe+subscribe will execute in the wrong order
         # because unsubscribe is done with help of async_create_task
-        if topic_changed:
-            for trig in self.trigger_instances:
-                await trig.async_attach_trigger()
+        if not topic_changed:
+            return
+        await asyncio.gather(
+            *[
+                create_eager_task(trig.async_attach_trigger())
```

Comment:
If the task never suspends it faster to await in series

Reply:
Reverted the asyncio.gather to await in series